manifests/config.pp

```diff
@@ -7,37 +7,10 @@
   }
 
   create_resources( 'jenkins::sysconfig', $::jenkins::config_hash )
-
-  $dir_params = {
-    ensure => directory,
-    owner  => $::jenkins::user,
-    group  => $::jenkins::group,
-    mode   => '0755',
-  }
-
-  # ensure_resource is used to try to maintain backwards compatiblity with
-  # manifests that were able to external declare resources due to the
-  # old conditional behavior of jenkins::plugin
-  if $::jenkins::manage_user {
-    ensure_resource('user', $::jenkins::user, {
-      ensure     => present,
-      gid        => $::jenkins::group,
-      home       => $::jenkins::localstatedir,
-      managehome => false,
-      system     => true,
-    })
+  if ! has_key($::jenkins::config_hash, 'JENKINS_HOME') {
```

Comment:
If a `Jenkins::Sysconfig['JENKINS_HOME']` resource is already defined, this will fail.  I don't think it is safe to assume that it won't happen as `jenkins::sysconfig` is a public interface (I'd rather we didn't have both `config_hash` and the defined type but we are stuck for the moment). 

I think using `defined` to check for the existence of the resource might be marginally safer?  That doesn't prevent the end user from causing a duplicate resource declaration by having a `JENKINS_HOME` key in `config_hash` and a `jenkins::sysconfig` resource but I feel it should be more obvious to the end user that they shouldn't define the same value twice.